Tools/TestWebKitAPI/Tests/WebCore/FontCascade.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ * Copyright (C) 2013 Google Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include <WebCore/FontCascade.h>
+
+namespace TestWebKitAPI {
+
+using CodePath = WebCore::FontCascade::CodePath;
+using FontCascade = WebCore::FontCascade;
+
+static void testCodePath(UChar codePoint, CodePath codePath)
+{
+    UChar target[1] = { codePoint };
+    EXPECT_EQ(codePath, WebCore::FontCascade::characterRangeCodePath(std::span<UChar>(std::span<UChar>(target)))) << "target: " << static_cast<int>(target[0]);
+}
+
+struct CodePathRange {
+    UChar start;
+    UChar end;
+    CodePath path;
+    CodePath belowPath { CodePath::Simple };
+    CodePath abovePath { CodePath::Simple };
+};
+
+static void testCodePathRange(CodePathRange range)
+{
+    ASSERT(range.start > 0);
+    UChar below[1] = { static_cast<UChar>(range.start - 1) };
+    UChar start[1] = { range.start };
+    UChar middle[1] = { static_cast<UChar>((range.start + range.end) / 2) };
+    UChar end[1] = { range.end };
+    UChar above[1] = { static_cast<UChar>(range.end + 1) };
+
+    EXPECT_EQ(range.belowPath, WebCore::FontCascade::characterRangeCodePath(std::span<UChar>(below))) << "below: " << std::hex << static_cast<int>(below[0]);
```

Comment:
nit: you probably don't need the `WebCore::` prefix anymore with the `using` statements above.

Reply:
That doesn't work: `using declaration cannot refer to class member`. I needed the alias.